tests/datasets/test_in_context_learning_datasets.py

```diff
@@ -0,0 +1,118 @@
+    assert 'mode' in next(dl.dataloader._get_iterator())
+
+
+@pytest.mark.parametrize('dataset_uri', ['piqa_small.jsonl'])
+def test_mc_task_dataloader(dataset_uri, tiny_gpt2_tokenizer):
+    local_data = os.path.join(os.path.dirname(__file__), 'local_data')
+
+    tokenizer = tiny_gpt2_tokenizer
+    dataset_uri = f'{local_data}/{dataset_uri}'
+    dl = get_mc_task_dataloader(dataset_uri,
+                                tokenizer,
+                                2,
+                                max_seq_len=2048,
+                                eos_tok_id=tokenizer.eos_token_id,
+                                num_fewshot=1,
+                                preamble_string='',
+                                example_delimiter='\n',
+                                continuation_delimiter=': ')
+
+    assert isinstance(dl.dataloader, DataLoader)  # pyright
+    assert 'input_ids' in next(dl.dataloader._get_iterator())
+    assert 'attention_mask' in next(dl.dataloader._get_iterator())
+    assert 'continuation_indices' in next(dl.dataloader._get_iterator())
+    assert 'labels' in next(dl.dataloader._get_iterator())
+    assert 'mode' in next(dl.dataloader._get_iterator())
+    assert 'gold_indices' in next(dl.dataloader._get_iterator())
+    assert 'choice_groupings' in next(dl.dataloader._get_iterator())
+
+
+@pytest.mark.parametrize('dataset_uri', ['lambada_small.jsonl'])
+@pytest.mark.parametrize('num_fewshot', [0, 5])
+@world_size(1, 2)
+@device('cpu', 'gpu')
+def test_lm_task_evaluation(device, world_size, dataset_uri, num_fewshot, tiny_gpt2_tokenizer):
+    in_memory_logger = InMemoryLogger()  # track the logged metrics in the in_memory_logger
+    local_data = os.path.join(os.path.dirname(__file__), 'local_data')
+    dataset_uri = f'{local_data}/{dataset_uri}'
+    tokenizer = tiny_gpt2_tokenizer
+    dl = get_lm_task_dataloader(dataset_uri,
+                                tokenizer,
+                                2,
+                                max_seq_len=2048,
+                                eos_tok_id=tokenizer.eos_token_id,
+                                num_fewshot=num_fewshot,
+                                preamble_string='',
+                                example_delimiter='\n',
+                                continuation_delimiter='')
+    evaluator = Evaluator(label='lambada', dataloader=dl, metric_names=['InContextLearningLMAccuracy'])
+    model = create_gpt2(use_pretrained=False, pretrained_model_name='EleutherAI/gpt-neo-125M')
+    model.add_eval_metrics(evaluator)
+    trainer = Trainer(model=model, max_duration='1ep', loggers=in_memory_logger)
+    trainer.eval(eval_dataloader=evaluator)
+    assert 'metrics/lambada/InContextLearningLMAccuracy' in in_memory_logger.data.keys()
+    assert in_memory_logger.data['metrics/lambada/InContextLearningLMAccuracy'][0][1].item() == 0
+
+
+@pytest.mark.parametrize('dataset_uri', ['piqa_small.jsonl', 'hellaswag_small.jsonl'])
```

Comment:
Is there a need to parametrize over the two datasets? They look of the same format to me.

Reply:
They test different number of answers per Q (PIQA has two choices, hellaswag 4)